### PR TITLE
GG-27581 .NET: Add partition-based AffinityCall and AffinityRun overloads

### PR DESCRIPTION
**Pull request by ptupitsyn**
* Add AffinityCall and AffinityRun overloads that take multiple cache names and partition
* Fix existing AffinityCall and AffinityRun: use proper Java APIs that reserve the partition instead of custom Task-based implementation

### CODE REVIEW

File: modules/platforms/dotnet/Apache.Ignite.Core/Compute/ICompute.cs [hunk 1]
```diff
@@ -532,6 +567,38 @@ public interface ICompute
         /// <returns>Task.</returns>
         Task AffinityRunAsync(string cacheName, object affinityKey, IComputeAction action, 
             CancellationToken cancellationToken);
+        
+        /// <summary>
+        /// Executes given job on primary node for the given partition.
+        /// (a.k.a. affinity co-location).
```
Comment (ashapkin) [inline modules/platforms/dotnet/Apache.Ignite.Core/Compute/ICompute.cs:573]: Let's be consistent with AffinityCall

File: modules/core/src/main/java/org/apache/ignite/internal/processors/platform/compute/PlatformBalancingSingleClosureAffinityTask.java [hunk 2]
```diff
@@ -31,6 +31,8 @@
 
 /**
  * Interop single-closure task with node balancing.
+ *
+ * NOTE: This class is not used, but preserved for compat during rolling upgrades.
```
Comment (ashapkin) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/platform/compute/PlatformBalancingSingleClosureAffinityTask.java:35]: This class doesn't transfer through the wire, why should we take care of it during RU?

File: modules/platforms/dotnet/Apache.Ignite.Core/Compute/ICompute.cs (hunk 1 above)
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite.Core/Compute/ICompute.cs:573]: Fixed

File: modules/core/src/main/java/org/apache/ignite/internal/processors/platform/compute/PlatformBalancingSingleClosureAffinityTask.java (hunk 2 above)
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/platform/compute/PlatformBalancingSingleClosureAffinityTask.java:35]: Good catch, fixed

**Pull request merged**